WebApplication/Backend/Services/AppointmentService.cs

```diff
@@ -98,13 +100,78 @@ private List<TimeIntervalDTO> GetAvailableAppointments(List<Appointment> appoint
             }
             return result;
         }
+
         private bool CompareTimeIntervals(DateTime dateTime1, DateTime dateTime2)
         {
             return dateTime1.Hour == dateTime2.Hour && dateTime1.Minute == dateTime2.Minute;
         }
+
+        public List<AppointmentWithRecommendationDTO> AppointmentRecomendationWithPhysicianPriority(string physicianId, string specializationName, string[] dates)
+        {
+            List<AppointmentWithRecommendationDTO> availableAppointments = new List<AppointmentWithRecommendationDTO>();
+            foreach (string date in dates)
+            {
+                availableAppointments.Add(new AppointmentWithRecommendationDTO(date, physicianId, GetAllAvailableAppointments(physicianId, specializationName, date), ""));
+            }
+            if (!availableAppointments.Any())
+            {
+                foreach (string date in AdditionalDates(dates))
+                {
+                    availableAppointments.Add(new AppointmentWithRecommendationDTO(date, physicianId, GetAllAvailableAppointments(physicianId, specializationName, date), ""));
+                }
+            }
+            return availableAppointments;
+        }
+
+        private List<String> AdditionalDates(string[] dates)
+        {
+           List<string> newDates = new List<string>();
+            DateTime dateTimeFrom = dateTimeDTO.CreateDateTime(dates[0]);
+            DateTime dateTimeTo = dateTimeDTO.CreateDateTime(dates[dates.Length - 1]);
+            while (newDates.Count < 3)
+            {
+                dateTimeTo = dateTimeTo.AddDays(1);
+                newDates.Add(DateConversion(dateTimeTo));
+            }
+            while (newDates.Count < 6 && dateTimeDTO.IsPreferredTimeValid(DateConversion(dateTimeFrom)))
+            {
+                dateTimeFrom = dateTimeFrom.AddDays(-1);
+                newDates.Add(DateConversion(dateTimeFrom));
+            }
+            return newDates;
+        }
+
+        private string DateConversion(DateTime date)
+        {
+            string[] dateString = date.ToString().Split(" ");
+            string[] partsOfDate = dateString[0].Split("/");
+            return partsOfDate[2] + "-" + partsOfDate[0] + "-" + partsOfDate[1];
```

Comment:
dateTime.toString("dd-MM-yyyy"); or whatever format you want
https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings